### PR TITLE
Upsolving after virtual contest

### PR DESCRIPTION
**Pull request by grikukan**
Small bugfixes (negative time in standings of virtual contests, invisible "Virtual Kirov" option when choosing contest type.

### CODE REVIEW

File: runlog.c [hunk 1]
```diff
@@ -312,6 +312,7 @@ run_add_record(
         int            eoln_type,
         int            variant,
         int            is_hidden,
+        int            upsolving,
```
Comment (blackav) [inline runlog.c]: This parameter is not necessary. The 'upsolving' value is not stored, but only used for parameter check.

File: runlog.c [hunk 2]
```diff
@@ -392,7 +393,7 @@ run_add_record(
       stop_time = ue->stop_time;
       if (!stop_time && state->head.duration)
         stop_time = ue->start_time + state->head.duration;
-      if (stop_time && timestamp > stop_time && state->head.duration) {
+      if (stop_time && timestamp > stop_time && state->head.duration && !upsolving) {
```
Comment (blackav) [inline runlog.c]: It's better to remove this parameter check from this function.

File: runlog.c (hunk 1 above)
Comment (grikukan) [inline runlog.c]: I don't see any nice way to check if user is upsolving inside this function. The only way I see to make it without additional parameter is to check it before calling run_add_record, but there are 7 usages of this function in code, so it won't be nice, I think.

Comment (grikukan) [inline runlog.c]: Maybe change runlog_state_t to serve_state_t and just make state = cs->runlog_state ?

Comment (blackav) [inline runlog.c]: I think that check in 'run_add_record' should be removed. The submission time check is performed before calling run_add_record in all cases anyway.

Comment (grikukan) [inline runlog.c]: #32438c3

Looks like there are some other similar checks, that are not required.

**Pull request merged**